### PR TITLE
[needs cherrypick to 6.2.*] Tests to upload SRPM - BZ#1378442

### PR DESCRIPTION
**Pull request by rplevka**
Tests to automate SRPM package upload:
https://bugzilla.redhat.com/show_bug.cgi?id=1378442

### CODE REVIEW

**Comment by rplevka** [comment]
addresses https://github.com/SatelliteQE/robottelo/issues/3846

**Comment by rplevka** [comment]
```bash
$ py.test -k test_positive_upload_content_srpm cli/test_repository.py
=================================================================================== test session starts ====================================================================================
platform linux2 -- Python 2.7.11, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/rplevka/work/rplevka/robottelo, inifile: 
plugins: xdist-1.14
collecting 0 items
=============================================================================== no tests ran in 0.72 seconds ===============================================================================

```
cli test is currently failing for 6.3 compose  due to the following error in stderr:
```bash
2016-09-22 14:52:40 - robottelo - WARNING - stderr contains following message:
[ERROR 2016-09-22 08:52:34 Modules] Error while loading module hammer_cli_import
[ERROR 2016-09-22 08:52:34 Modules] <Gem::ConflictError> Unable to activate hammer_cli_import-0.11.2, because hammer_cli_katello-0.1.0 conflicts with hammer_cli_katello (~> 0.0.6)
```

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -646,6 +647,25 @@ def test_positive_update_contents(self):
         # Verify the repository's contents.
         self.assertEqual(repo.read().content_counts['rpm'], 1)
 
+    @skip_if_bug_open('bugzilla', 1378442)
+    @run_only_on('sat')
+    @tier2
```
Comment (omaciel) [inline tests/foreman/api/test_repository.py]: I'd argue that this should be `tier1`

File: tests/foreman/api/test_repository.py [hunk 2]
```diff
@@ -646,6 +647,25 @@ def test_positive_update_contents(self):
...
+    def test_positive_update_contents_srpm(self):
```
Comment (omaciel) [inline tests/foreman/api/test_repository.py]: s/update/upload?

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -919,6 +920,29 @@ def test_positive_upload_content(self):
...
+            'name': new_repo['name'],
+            'organization': new_repo['organization'],
+            'path': "/tmp/{0}".format(SRPM_TO_UPLOAD),
+            'product-id': new_repo['product']['id'],
+        })
+        self.assertIn(
+            "Successfully uploaded file '{0}'".format(SRPM_TO_UPLOAD),
+            result[0]['message'],
+        )
+
```
Comment (omaciel) [inline tests/foreman/cli/test_repository.py]: To be consistent with the API test, how about asserting that you can list the new package as well?

File: tests/foreman/api/test_repository.py [hunk 4]
```diff
@@ -646,6 +647,25 @@ def test_positive_update_contents(self):
...
+        """Create a repository and upload SRPM contents.
+
+        @id: e091a725-048f-44ca-90cc-c016c450ced9
+
+        @Assert: The repository's contents include one SRPM.
+
+        @CaseLevel: Integration
+        """
+        # Create a repository and upload RPM content.
```
Comment (sghai) [inline tests/foreman/api/test_repository.py]: RPM/SRPM ?

**Comment by sghai** [comment]
I see test implementation from CLI & API. Are your planning to add test from UI too ?

**Comment by omaciel** [comment]
@sghai I updated my issue https://github.com/SatelliteQE/robottelo/issues/3846 so that we can handle UI as well.

File: tests/foreman/cli/test_repository.py (hunk 3 above)
Comment (rplevka) [inline tests/foreman/cli/test_repository.py]: that'd require implementing that in `nailgun` first. For now, it only supports listing the content counts

Comment (omaciel) [inline tests/foreman/cli/test_repository.py]: This is for CLI so if hammer does not support this, then we can skip it.

File: tests/foreman/api/test_repository.py (hunk 1 above)
Comment (omaciel) [inline tests/foreman/api/test_repository.py]: Don't forget this one :)

File: tests/foreman/api/test_repository.py [hunk 5]
```diff
@@ -646,6 +647,25 @@ def test_positive_update_contents(self):
...
+    @run_only_on('sat')
+    @tier2
+    def test_positive_upload_contents_srpm(self):
+        """Create a repository and upload SRPM contents.
+
+        @id: e091a725-048f-44ca-90cc-c016c450ced9
+
+        @Assert: The repository's contents include one SRPM.
+
+        @CaseLevel: Integration
```
Comment (oshtaier) [inline tests/foreman/api/test_repository.py]: and don't forget to remove that

**Comment by oshtaier** [comment]
ACK pending comments

**Comment by omaciel** [comment]
By the way, I have filed #3852, #3853 and #3854 :)

**Comment by rplevka** [comment]
comments addressed

**Comment by sghai** [comment]
ACK

**Pull request merged**